### PR TITLE
separate layers with hot and cold sources, refs #296

### PR DESCRIPTION
**Pull request by samanpwbb**
cc: @mourner @mcwhittemore @lucaswoj 

I also removed the fallback layer id assignment. Ids for style layers should be required.

I'll add tests if this makes sense to everyone.

### CODE REVIEW

File: src/options.js [hunk 1]
```diff
@@ -24,6 +22,15 @@ const hideControls = {
   trash: false
 };
 
+function addSources(styles, temp) {
```
Comment (mcwhittemore) [inline src/options.js]: Lets call `temp` something like `sourceBucket`

Comment (samanpwbb) [inline src/options.js]: 🔥 ❄️

**Comment by mcwhittemore** [comment]
This looks great @samanpwbb

**Comment by samanpwbb** [comment]
Going to write some tests then merge 👍

**Comment by samanpwbb** [comment]
Added tests, merging when green.

**Pull request merged**